test/yz_index_hashtree_eqc.erl

```diff
@@ -0,0 +1,553 @@
+        eqc_gen:gen(eqc_statem:call()).
+start_kv_tree_command(_S) ->
+    {call, ?MODULE, start_kv_tree, []}.
+
+%% @doc start_kv_tree_pre - Precondition for generation
+-spec start_kv_tree_pre(S :: eqc_statem:symbolic_state()) -> boolean().
+start_kv_tree_pre(S) ->
+    S#state.kv_idx_tree == undefined.
+
+%% @doc start_kv_tree_next - Next state function
+-spec start_kv_tree_next(S :: eqc_statem:symbolic_state(),
+                         V :: eqc_statem:var(),
+                         Args :: [term()]) -> eqc_statem:symbolic_state().
+start_kv_tree_next(S, V, _Args) ->
+    S#state{kv_idx_tree=V, kv_idx_objects=dict:store({0, 1}, dict:new(),
+                                               S#state.kv_idx_objects)}.
+
+%% @doc start_kv_tree_post - Postcondition for start_kv_tree
+-spec start_kv_tree_post(S :: eqc_statem:dynamic_state(),
+                         Args :: [term()], R :: {ok, tree()}) -> true | term().
+start_kv_tree_post(_S, _Args, R) ->
+    case R of
+        {ok, TreePid} ->
+            is_process_alive(TreePid);
+        _ ->
+            error
+    end.
+
+%% @doc Spawns a kv idx hashtree gen_server process with a default `Index`
+-spec start_kv_tree() -> {ok, tree()}.
+start_kv_tree() ->
+    riak_kv_index_hashtree:start_link(0, ?MODULE, []).
+
+%% ------ Grouped operator: insert_yz_tree
+%% @doc insert_yz_tree_command - Command generator
+-spec insert_yz_tree_command(S :: eqc_statem:symbolic_state()) ->
+        eqc_gen:gen(eqc_statem:call()).
+insert_yz_tree_command(S) ->
+    {call, ?MODULE, insert_yz_tree, [insert_method(), eqc_util:riak_object(),
+                                     S#state.yz_idx_tree]}.
+
+%% @doc insert_yz_tree_pre - Precondition for insert_yz_tree
+-spec insert_yz_tree_pre(S :: eqc_statem:symbolic_state(),
+                         Args :: [term()]) -> boolean().
+insert_yz_tree_pre(S, _Args) ->
+    S#state.yz_idx_tree /= undefined.
+
+%% @doc insert_yz_tree_next - Next state function
+-spec insert_yz_tree_next(S :: eqc_statem:symbolic_state(),
+                          V :: eqc_statem:var(),
+                          Args :: [term()]) -> eqc_statem:symbolic_state().
+insert_yz_tree_next(S, _V, [_, RObj, _]) ->
+    {ok, TreeData} = dict:find({0, 1}, S#state.yz_idx_objects),
+    S#state{yz_idx_objects=dict:store({0, 1},
+                                      set_treedata(RObj, TreeData),
+                                      S#state.yz_idx_objects),
+            trees_updated=false}.
+
+%% @doc insert_yz_tree_post - Postcondition for insert_yz_tree
+-spec insert_yz_tree_post(S :: eqc_statem:dynamic_state(),
+                          Args :: [term()], R :: ok) -> true | term().
+insert_yz_tree_post(_S, _Args, _Res) ->
+    true.
+
+-spec insert_yz_tree(sync|async, obj(), {ok, tree()}) -> ok.
+insert_yz_tree(Method, RObj, {ok, TreePid}) ->
+    BKey = eqc_util:get_bkey_from_object(RObj),
+    yz_index_hashtree:insert(Method, {0, 1}, BKey,
```

Comment:
For the future when this property gets further generalized, I'd love to see this `{0,1}` tuple be generated (or derived from the data). Consider replacing the tuple with a macro so it is easily understood that it is a "magic constant" in this version.